apps/graphql/src/dataloaders/itinerariesHelpers.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+
+import * as DateFNS from 'date-fns';
+
+import type { ApiRouteItem, Sector } from './ItinerariesloaderTypes';
+
+export const differenceInMinutes = (from: ?string, to: ?string) => {
+  return from && to ? DateFNS.differenceInMinutes(to, from) : null;
+};
+
+export const mapVehicle = (type: ?string, uniqueNo: ?string) => ({
+  type: type ?? null,
+  uniqueNo: uniqueNo ?? null,
+});
+
+export const mapTransporter = (transporter: ?string) => ({
+  name: transporter ?? null,
+});
+
+export const mapItineraryType = (routes: ?Array<Array<string>>) => {
+  if (routes == null) {
+    return null;
+  }
+  if (routes.length === 1) {
+    return 'oneway';
+  }
+  if (routes.length === 2) {
+    return 'return';
+  }
+  return null;
+};
+
+export const sortRoute = (
+  segments: Array<ApiRouteItem>,
+): Array<ApiRouteItem> => {
+  return segments.slice(0).sort((segmentA, segmentB) => {
+    const a = segmentA.utc_departure && new Date(segmentA.utc_departure);
+    const b = segmentB.utc_departure && new Date(segmentB.utc_departure);
+    if (!a || !b) return 0;
+    return a < b ? -1 : a > b ? 1 : 0;
+  });
+};
+
+export const mapSectors = (
```

Comment:
Will be possible to add some explanation comment to this function because it is really hard to understand what's going on there

Reply:
This is not what I mean 😃. 
Try to imagine that you will look at the `mapSectors` after a few mounts.
I really don't understand what's going on. 
Try to explain what happens in first map iteration on routesMaps, this is the most confusing part. thanks

Reply:
Done